src/main/java/org/spongepowered/asm/mixin/MixinEnvironment.java

```diff
@@ -629,7 +629,7 @@ final String getStringValue() {
         <E extends Enum<E>> E getEnumValue(E defaultValue) {
             String value = System.getProperty(this.property, defaultValue.name());
             try {
-                return (E)Enum.valueOf(defaultValue.getClass(), value.toUpperCase(Locale.ROOT));
+                return (E)Enum.valueOf(defaultValue.getDeclaringClass(), value.toUpperCase(Locale.ROOT));
```

Comment:
The cast to `E` and the `@SuppressWarnings` can both be removed, `getDeclaringClass` returns `Class<E>` rather than `Class<?>`